types/weak-napi/weak-napi-tests.ts

```diff
@@ -0,0 +1,26 @@
+import weak = require('weak-napi');
+
+const obj = {a: 123};
+
+const weakReference = weak(obj, () => {
+    // collected
+});
+
+const sameType = weak.get(weakReference);
+
+function foo(a: {a: number}) {}
+
+if (sameType) {
+    foo(sameType);
+}
+
+const anyVar = null as any;
+
+if (weak.isWeakRef(anyVar)) {
+    const a = anyVar; // WeakRef<any>
+}
+
+if (weak.isDead(weakReference)) {
+    const a = weakReference; // WeakRef<undefined>
```

Comment:
FYI you can use `// $ExpectType WeakRef<undefined>` and it’ll actually be validated as such

Reply:
@andrewbranch I updated it